iFixit/react-commerce
# Add product list hreflangs

Satisfies https://github.com/iFixit/ifixit/issues/50649 for the headless store.

As mentioned here https://github.com/iFixit/ifixit/issues/50649#issuecomment-1816735791, if the customer has an item type selected on a product list page that should have hreflangs, we exclude the hreflangs (unless we specifically have hreflangs set for that device/item type combination).

## CR

Please carefully review the commit that prunes redundant mappings (0d55b0f9546fd7c06571122647cdc219ca0aa14c), as I did this by eye. My intent was to reduce the bundle size increase and improve readability.

## Bundle size increase

The product list pages have a ~1 kb bundle size increase due to the size of the hreflang mappings.

## QA

Compare the [spec spreadsheet](https://docs.google.com/spreadsheets/d/1bFai7oXL1HJtgFm69taKA_VjKN8WtXyrJfMyB6UQ7xY/edit#gid=1714817208) with the hreflang tags present in the `<head>` on a variety of product list pages.

Make sure to cover the categories of `/Tools/*`, `/Parts/*`, `/Parts`, and the only item type hreflang (`/iMac/Kits`). Also make sure that the hreflangs do not show up when they aren't supposed to.
